### DISCUSSION TITLE
why is option.none not of type option.None?

### DISCUSSION THREAD

**Discussion by pbadenski** [gcanti/fp-ts#1587] [Q&A]
`const success: option.None = option.none;`

doesn't compile and the error is:
```
TS2322: Type 'Option<never>' is not assignable to type 'None'.
  Type 'Some<never>' is not assignable to type 'None'.
    Types of property '_tag' are incompatible.
      Type '"Some"' is not assignable to type '"None"'
```

Is this by design? Just trying to wrap my brains around it....

**Comment by DenisFrezzato** [comment]
`none` and `some` are seen as data constructors for the data type `Option<A>`, so both returns a new `Option<A>`. The purpose of data constructors is to create a new value of their data type. This applies for any data type and their respective data constructors. You should never deal with the type of the data constructors, but with the data type.

**Comment by pbadenski** [comment]
Why are `None` and `Some` interfaces exported then?

**Reply by enricopolanski** [reply]
Seems like a good point, I checked most of the fp-ts codebase and I can't find places where they are imported.

**Comment by waynevanson** [comment]
`const success: option.None = option.none;` is one of two types: `Some<A>` and `None`.

You'll need to narrow the type via an assetion:
`const success: option.None = option.none as option.None`
 or 
`const success = option.none as option.None`
